packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -220,8 +179,36 @@ class GrpcGatewayListener extends VirtualGatewayListener {
           port: this.port,
           protocol: Protocol.GRPC,
         },
-        healthCheck: this.healthCheck? this.renderHealthCheck(this.healthCheck): undefined,
+        healthCheck: this.healthCheck ? renderHealthCheck(this.healthCheck, this.protocol, this.port): undefined,
       },
     };
   }
-}
\ No newline at end of file
+}
+
+function renderHealthCheck(
+  hc: HealthCheck, listenerProtocol: Protocol, listenerPort: number): CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty {
```

Comment:
I don't know if this is the proper way to break up a function that violates the 150 max len lint rule.

Reply:
Seems like the linter accepts it, so should be good 🙂